### PR TITLE
[iterator.requirements.general] Rephrase definition of 'writable to'

### LINKED ISSUE #698: Fix the definition of "writable to"
The definition of "writable" to is problematic as Richard explains below, but also the definition should be for "writable to", not just "writable", and it should be indexable.

e-mail from Richard:

> It's defined in [iterator.requirements.general]/1: "All output iterators support the expression *i = o where o is a value of some type that is in the set of types that are writable to the particular iterator type of i."
> 
> That said, this wording seems fairly poor: it suggests that *i = o only has to work for one type in the set of "writable to" types, not for _all_ such types, and is backwards: this is defining what "writable to" means, not adding a constraint on output iterators. Perhaps it would be clearer as:
> 
> "All output iterators i have a set of types that are writable to the iterator; for each such type T, the expression *i = o is valid where o is an expression of type T."

### PR DESCRIPTION
**Pull request by jensmaurer**
for output iterators.

Fixes #698.

### CODE REVIEW

File: source/iterators.tex [hunk 1]
```diff
@@ -46,14 +46,10 @@
...
 of the iterator.
-All output iterators support the expression
-\tcode{*i = o}
-where
-\tcode{o}
-is a value of some type that is in the set of types that are
-\term{writable}
-to the particular iterator type of
-\tcode{i}.
+All output iterators \tcode{i} have a non-empty set of types that are
```
Comment (tkoeppe) [inline source/iterators.tex]: I don't think the plural works here. How about "An output iterator `i` has a non-empty set..."?

**Comment by jensmaurer** [comment]
I've pushed an update that addresses your concern, @tkoeppe.  I do want to point out that the surrounding text uses "All ... iterators...", although I do agree singular would be better everywhere.

**Comment by tkoeppe** [comment]
Indeed it does... maybe we can change those ambient uses, too? They sound pretty weird at the moment. Cross-check with C++98 if this got grandfathered in; I'd be happy to improve the text at this point.

**Comment by jensmaurer** [comment]
Here you are. I haven't checked C++98 yet, but the LaTeX there was one-word-per-line, which suggests troff heritage.

**Comment by tkoeppe** [comment]
Thanks, much better!

**Pull request merged**